From f906e3df172e07ac82cdd87b278d7896949262ea Mon Sep 17 00:00:00 2001
From: Rabeeh Khoury <rabeeh@solid-run.com>
Date: Mon, 2 Nov 2015 18:19:32 +0200
Subject: [PATCH 3/3] clearfog: reset usom onboard 1512 phy
Organization: O.S. Systems Software LTDA.

Use GPIO19 which is wired to the usom phy reset signal in order to reset the
on usom 1512 gigabit Ethernet phy.

This GPIO is valid on ClearFog rev 2.1 and newer.

Signed-off-by: Rabeeh Khoury <rabeeh@solid-run.com>
---
 board/mv_ebu/a38x/armada_38x_family/boardEnv/mvBoardEnvSpec38x.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/board/mv_ebu/a38x/armada_38x_family/boardEnv/mvBoardEnvSpec38x.c b/board/mv_ebu/a38x/armada_38x_family/boardEnv/mvBoardEnvSpec38x.c
index ba62ca5..bfcc4c4 100644
--- a/board/mv_ebu/a38x/armada_38x_family/boardEnv/mvBoardEnvSpec38x.c
+++ b/board/mv_ebu/a38x/armada_38x_family/boardEnv/mvBoardEnvSpec38x.c
@@ -296,8 +296,12 @@ void a38x_customer_board_1_gpp_callback(MV_BOARD_INFO *board) {
 	/* Toggle GPIO41 to reset onboard switch and phy */
 	MV_REG_BIT_RESET (0x18140, 0x200);
 	MV_REG_BIT_RESET (0x18144, 0x200);
+	/* GPIO 19 on ClearFog rev 2.1 controls the uSOM onboard phy reset */
+	MV_REG_BIT_RESET (0x18100, 0x00080000);
+	MV_REG_BIT_RESET (0x18104, 0x00080000);
 	mvOsDelay(1);
 	MV_REG_BIT_SET (0x18140, 0x200);
+	MV_REG_BIT_SET (0x18100, 0x00080000);
 	mvOsDelay(10);
 	for (i = 0; i < board->numIoExp; i++) {
 		if (MV_OK != mvBoardTwsiSet(BOARD_TWSI_IO_EXPANDER, board->pIoExp[i].addr,
-- 
2.7.1

